docs/sql/sql.md

```diff
@@ -0,0 +1,440 @@
+SELECT * FROM Inventory;
+
+-- Select the names of the items in my inventory
+SELECT item_name FROM Inventory
+
+-- How many items are in my inventory?
+SELECT COUNT(*) as num_items FROM Inventory
+```
+
+#### FROM Clause
+
+```ebnf
+FROM table [ [AS] alias ] { [INNER] JOIN table [ [AS] alias ] ON predicate }
+```
+
+The query language supports joins among an arbitrary number of tables.
+It does not impose any index restrictions like the subscription language.
+
+Examples:
+```sql
+-- Which orders include currently stocked items?
+SELECT order.*
+FROM Orders order
+JOIN Inventory inv
+ON order.item_id = inv.item_id
+```
+
+#### WHERE Clause
+
+```ebnf
+predicate
+    = expr
+    | predicate AND predicate
+    | predicate OR  predicate
+    ;
+
+expr
+    = literal
+    | column
+    | expr op expr
+    ;
+
+op
+    = '='
+    | '<'
+    | '>'
+    | '<' '='
+    | '>' '='
+    | '!' '='
+    | '<' '>'
+    ;
+
+literal
+    = INTEGER
+    | FLOAT
+    | STRING
+    | HEX
+    | TRUE
+    | FALSE
+    ;
+```
+
+The query language supports both conjunctions and disjunctions.
+It also supports `!=` comparisons.
+It does not support arithmetic expressions.
+
+Examples:
+```sql
+SELECT * FROM Inventory WHERE item_id = 1;
+```
+
+#### ORDER BY Clause
+
+```ebnf
+ORDER BY column [ ASC | DESC ] { ',' column [ ASC | DESC ] }
+```
+
+`ORDER BY` sorts the result set by one or more of the projected columns.
+Each column involved must be of a type that allows comparison.
+Ascending is the default behavior if not qualified.
+
+Examples:
+```sql
+SELECT * FROM Inventory ORDER BY price;
+```
+
+#### LIMIT Clause
+
+```ebnf
+LIMIT INTEGER
+```
+
+`LIMIT` restricts the number of rows returned.
+
+Examples:
+```sql
+-- What are the 5 highest priced items in my inventory?
+SELECT * FROM Inventory ORDER BY price LIMIT 5;
+```
+
+### INSERT
+
+```ebnf
+INSERT INTO table [ '(' column { ',' column } ')' ] VALUES '(' literal { ',' literal } ')'
+```
+
+Examples:
+```sql
+-- Inserting one row
+INSERT INTO Inventory (item_id, item_name) VALUES (1, 'health1');
+
+-- Inserting two rows
+INSERT INTO Inventory (item_id, item_name) VALUES (1, 'health1'), (2, 'health2');
+```
+
+### DELETE
+
+```ebnf
+DELETE FROM table [ WHERE predicate ]
+```
+
+Deletes all rows from a table.
+If `WHERE` is specified, only the matching rows are deleted.
+
+Examples:
+```sql
+-- Delete all rows
+DELETE FROM Inventory;
+
+-- Delete all rows with a specific item_id
+DELETE FROM Inventory WHERE item_id = 1;
+```
+
+### UPDATE
+
+```ebnf
+UPDATE table SET [ '(' assignment { ',' assignment } ')' ] [ WHERE predicate ]
+```
+
+Updates column values of existing rows in a table.
+The columns are identified by the `assignment` defined as `column '=' expr`.
+The column values are updated for all rows that match the `WHERE` condition.
+The rows are updated after the `WHERE` condition is evaluated for all rows.
+
+Examples:
+```sql
+-- Update the item_name for all rows with a specific item_id
```

Comment:
```suggestion
-- Update all the rows
UPDATE Inventory SET item_name = 'new name';
-- Update the item_name for all rows with a specific item_id
```

Reply:
I'll leave this one for now, just because updating all rows is less common. But it is specified in the language.